pkgs/by-name/up/upcloud-cli/package.nix

```diff
@@ -0,0 +1,49 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  versionCheckHook,
+}:
+
+buildGoModule rec {
+  pname = "upcloud-cli";
+  version = "3.14.0";
+
+  src = fetchFromGitHub {
+    owner = "UpCloudLtd";
+    repo = "upcloud-cli";
+    tag = "v${version}";
+    hash = "sha256-zKPoJFfgqi6ZIeZKJy7YeYuqHWVPH0LXvWpOYCEM7dE=";
+  };
+
+  vendorHash = "sha256-76bLk4zten9SGXbt/M8VKPSylCwQqclyscSVQQaAtbA=";
+
+  ldflags = [
+    "-s -w -X github.com/UpCloudLtd/upcloud-cli/v3/internal/config.Version=${version}"
+  ];
+
+  subPackages = [
+    "cmd/upctl"
+    "internal/*"
+  ];
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  versionCheckProgram = "${placeholder "out"}/bin/upctl";
+  versionCheckProgramArg = [ "version" ];
+  doInstallCheck = true;
+
+  passthru = {
+    updateScript = lib.nix-update-script { };
```

Comment:
This updateScript doesn't eval, fixed in https://github.com/NixOS/nixpkgs/pull/372805.